src/name/ip_address.rs

```diff
@@ -537,12 +579,291 @@ mod tests {
 
     #[test]
     fn ipv6_octets_test() {
+        // Invalid UTF-8 encoding
         assert_eq!(
-            ipv6_octets(b"2a05:d018:076c:b684:8e48:47c9:84aa:b34d"),
-            Ok([
-                0x2a, 0x05, 0xd0, 0x18, 0x07, 0x6c, 0xb6, 0x84, 0x8e, 0x48, 0x47, 0xc9, 0x84, 0xaa,
-                0xb3, 0x4d
-            ])
+            ipv6_octets(b"\xc3\x28a05:d018:076c:b684:8e48:47c9:84aa:b34d"),
+            Err(InvalidIpAddressError),
         );
+        // Invalid hexadecimal
+        assert_eq!(
+            ipv6_octets(b"ga05:d018:076c:b684:8e48:47c9:84aa:b34d"),
+            Err(InvalidIpAddressError),
+        );
+    }
+
+    #[test]
```

Comment:
This test should be marked `#[cfg(feature = "std")]` like the function it tests.

Reply:
Ugh, sorry, again c&p shenanigan.

Reply:
I have updated the PR. I think it's better to move the function to the `alloc_tests` `mod` as I have updated this PR, so all tests that require allocation are within this module.

Reply:
Ah, that is a better idea. Have incorporated that into feat-ip-address